src/Microsoft.AspNetCore.Authorization/AuthorizationHandler.cs

```diff
@@ -6,38 +6,53 @@
 
 namespace Microsoft.AspNetCore.Authorization
 {
+    /// <summary>
+    /// Base class for authorization handlers that need to be called for a specific requirement type.
+    /// </summary>
+    /// <typeparam name="TRequirement">The type of the requirement to handle.</typeparam>
     public abstract class AuthorizationHandler<TRequirement> : IAuthorizationHandler
-        where TRequirement : IAuthorizationRequirement
+            where TRequirement : IAuthorizationRequirement
     {
-        public void Handle(AuthorizationHandlerContext context)
+        async Task IAuthorizationHandler.HandleAsync(AuthorizationHandlerContext context)
         {
             foreach (var req in context.Requirements.OfType<TRequirement>())
             {
-                Handle(context, req);
+                await HandleAsync(context, req);
             }
         }
 
-        public virtual async Task HandleAsync(AuthorizationHandlerContext context)
+        /// <summary>
+        /// Makes a decision if authorization is allowed based on a specific requirement.
+        /// </summary>
+        /// <param name="context">The authorization information.</param>
+        /// <param name="requirement">The requirement to evaluate.</param>
+        protected virtual void Handle(AuthorizationHandlerContext context, TRequirement requirement)
```

Comment:
I don't disagree but this is a much bigger breaking change as as of today this was the one that you HAD to implement. I also agree as the interface only expose the async one. But from @HaoK this class is also a way to provide a sugar around the sync version of Handle.

I really think that if the purpose of the class was clear, there wouldn't be any question about which methods to expose:
1- It's from filtering `TRequirement` then we should only expose `Handle(context, requirement)`
2- If it's to expose a synchronous method, then we need `Handle(context, requirement)`
3- Can be both